packages/apollo-server-core/src/requestPipelineAPI.ts

```diff
@@ -50,13 +52,17 @@ export interface GraphQLRequestContext<TContext = Record<string, any>> {
   readonly queryHash?: string;
 
   readonly document?: DocumentNode;
+  readonly originalDocumentString?: string;
```

Comment:
I'm still not really sure this is what we want, because the cache key shouldn't depend on the whole document if it contains more than one operation. So ideally we'd have a well-defined uniform notion of an `operationID`. But even if we do want to depend on the whole document for now, couldn't we use the `queryHash` instead? (We should probably rename that to `documentHash`.)

Reply:
This also only gets used for parse failures. But it gets sent to Engine and used to display the (failing) query in the UI. So it needs to be text, not a hash.

Reply:
But I'm going to call it originalDocumentText for now.

Re caching, which is a separate PR for now but the idea here, I think we're being strict enough about "this is a full request cache" and ignoring other semantically equivalent changes that "you have to be consistent about what the whole document looks like" seems like a fair place to start for v1 at least.

Reply:
Or maybe just documentText, since I think that captures the "original" concept to some degree.

Reply:
Yep, I like `documentText`!